File: BTaskServer/model/TaskItem.go
package model

import (
	"time"
)

type TaskItem struct {
	ID            uint      `gorm:"primaryKey" json:"id"`
	TaskID        string    `gorm:"unique;not null" json:"taskId"`
	TaskName      string    `gorm:"not null" json:"taskName"`
	Supplier      string    `json:"supplier"`
	UserTypes     string    `json:"userTypes"` // Consider using a more structured type if complex logic is needed
	PriceReal     float64   `json:"priceReal"`
	PriceProtocol float64   `json:"priceProtocol"`
	PriceManual   float64   `json:"priceManual"`
	SelectionMode string    `json:"selectionMode"`
	Status        string    `json:"status" gorm:"default:'active'"`
	CreatedAt     time.Time `json:"createdAt"`
	UpdatedAt     time.Time `json:"updatedAt"`
}

File: frontend/src/types/taskItem.ts
export interface TaskItem {
  id?: number;
  taskId: string;
  taskName: string;
  supplier: string;
  userTypes: string;
  priceReal: number;
  priceProtocol: number;
  priceManual: number;
  selectionMode: string;
  status: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface TaskItemFormData extends Omit<TaskItem, 'id' | 'createdAt' | 'updatedAt'> {
  // Any specific form-only fields or modifications
}

export interface TaskItemRes {
  list: TaskItem[];
  total: number;
} 